### PR TITLE
EQ to dowstream system format definition v0.0.1

### PR DESCRIPTION
**Pull request by dhilton**
__What__

This PR creates the documentation defining the data exchange format for a response from
a respondent. This commit also adds the ability to link that documentation into the Sphinx
generated documentation system.

A general introduction to the wider EDC system is also included for completeness.

__How to test__

1. Review the documents added.
2. Comment on the files, correcting any inaccuracies or misunderstandings.
3. Once happy - click merge.

__Who can review__

Anyone but @dhilton

### CODE REVIEW

File: docs/dx-eq.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+.. code-block:: javascript
+
+    {
+      "survey_id": "021",
+      "collection":{
+        "exercise_sid": "hfjdskf",
+        "instrument_id": "yui789",
+        "period": "2016-02-01"
+      },
+      "version":"0.0.1",
```
Comment (collisdigital) [inline docs/dx-eq.rst]: suggest the version, origin etc. are the first things in the file and add a "type" as we've discussed before, suggest was to use MIME to define it but I'm not expert (proposing some kind of logical namespace, but wondering if GDS et al have one already we should use??) e.g.

```javascript
{
    "type" : "application/uk.gov.ons.edc.eq.submission+json",
    "version" : "0.0.1",
    "origin" : "uk.gov.ons.edc.eq"
...
}
```

Comment (collisdigital) [inline docs/dx-eq.rst]: Looking deeper, and in the absence of any other expertise it seems vendor specific mimetypes should start with vnd if registered or we can use 'prs' for personal/unregistered. e.g. "prs.ukgovernment.ons.edc.eq.submission+json" see http://www.iana.org/assignments/media-types/media-types.xhtml 

Another alternative (or addition) would be to use something like a schema statement e.g. 

```javascript
    "$schema" : "http://ons.gov.uk/schema/edc/eq/submissionSchema"
```

Just throwing ideas out here!

**Comment by iwootten** [comment]
Not having yet been exposed to the downstream formats, I don't have many thoughts on this - however, this may be useful regarding namespacing/versioning: http://json-ld.org/

**Comment by davidcarboni** [comment]
I'm not a fan of custom mime types, feels like they add complexity and move away from plain get/post requests. I'd imagine a simple, readable identifier included in the data itself.

The website has a human-readable "type" field in the json so the UI can render the right template for the data, e.g. "bulletin":

https://www.ons.gov.uk/employmentandlabourmarket/peopleinwork/labourproductivity/bulletins/labourproductivity/q32015/data

**Comment by dhilton** [comment]
We could adopt a namespaced type identifier? `ons:surveyresponse` We could then in EQ create a `ons:surveyschema` type to match the response to, even linking the two via json-ld as @iwootten suggested.

Once a public survey-registry exists; you could link from the individual survey response to the URI of the survey schema, giving you a clear traceability from capture data (which most likely will end up being stored on disk / object-store / HDFS / etc) 

That approach could solve the basic cases and seems the most simple that "works for now" ?

**Comment by dhilton** [comment]
After internal discussions we settled the type example for now by namespacing type and adding version. We will review this in the future.

**Pull request merged**